server/webapp/WEB-INF/rails.new/spec/new_javascripts/models/shared/version_updater_spec.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define(['mithril','models/shared/version_updater'], function (m, VersionUpdater) {
+  describe("VersionUpdater", function () {
+    describe('update', function() {
+      it('should fetch the stale version info', function () {
+        var deferred = $.Deferred();
+        var xhr = jasmine.createSpyObj(xhr, ['setRequestHeader']);
+        var thirtyOneMinutesBack = new Date(Date.now() - 31 * 60 * 1000);
+
+        spyOn(localStorage, 'getItem').and.returnValue(JSON.stringify({last_updated_at: thirtyOneMinutesBack}));
+        spyOn(m, 'request').and.returnValue(deferred.promise());
```

Comment:
Instead of stubbing `m.request` any possibility of directly using mock-ajax — makes for cleaner behaviorial tests. Here you're effectively testing what attributes you set on the deferred, aka — the internal implementation of `m.request`. If that changes, we'd have to re-write a whole bunch of tests, in addition to changing the actual code.

See [this](https://github.com/gocd/gocd/blob/5de3e1bc8b8868fb9ad931501d6f27d6b6a6dd3d/server/webapp/WEB-INF/rails.new/spec/new_javascripts/models/elastic_profiles/elastic_profiles_spec.js#L90-L118) for reference.